common/src/worldgen.rs

```diff
@@ -62,13 +65,31 @@ impl NodeStateRoad {
     }
 }
 
+/// Contains a minimal amount of information about a node that can be deduced entirely from
+/// the NodeState of its parents.
+pub struct PartialNodeState {
```

Comment:
Maybe `InheritedNodeState`?

Reply:
I don't think `InheritedNodeState` fits here, since any horosphere the node itself generates also shows up here, so I probably need to reword the comment.

I need to somehow capture the concept that the full `NodeState` has to be generated in two parts, starting with a `PartialNodeState` and eventually upgrading into a full `NodeState`, and interdependencies between nodes mean that this cannot be fully abstracted away. Maybe `FirstPassNodeState` and `SecondPassNodeState`?

The splitting of node state in a readable way is something I think is worth bikeshedding, and I am interested in trying to make this work as well as possible.